### ISSUE TITLE
CrossEntropyLoss should not be used with one-hot encoded labels

### ISSUE THREAD

**Issue by patrontheo** [IGNF/FLAIR-1#9]
I think that you use CrossEntropyLoss with one-hot targets, but it is not supposed to be used like that. The code works because, as explained in the link below, the CrossEntropyLoss also accepts targets of size [N,C,...] in case it is class probabilities (it is not our case here).
https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

Some discussion talking about that:
https://discuss.pytorch.org/t/cross-entropy-with-one-hot-targets/13580/4
https://discuss.pytorch.org/t/which-loss-function-for-one-hot-encoded-labels/29920/2
https://discuss.pytorch.org/t/2d-crossentropyloss-for-one-hot-targets/117545/2

So I think you should use `targets = torch.argmax(targets, dim=1)` before computing the loss, or probably better to change directly in the dataloader.

FYI, we reached a mIoU of 56.22 with baseline model (no aug, no bottom).

**Comment by agarioud** [comment]
Hello @patrontheo, apologies for the delay in my response. 

The issue originated from a previous code version where labels were stored differently..
Modification in task_module to pass the argmax of targets to the cross-entropy loss function have been made. 

However, with multiple runs (with augmentations), I haven’t observed substantial improvements in the results. 
Nonetheless, thank you for your vigilance !
